autofaiss/indices/distributed.py

```diff
@@ -183,42 +188,55 @@ def _merge_from_local(merged: Optional[faiss.Index] = None) -> faiss.Index:
                 pbar.update(1)
 
     if batch_id is not None and tmp_output_folder is not None:
-        _save_small_index(
-            index=merged_index, batch_id=batch_id, small_indices_folder=tmp_output_folder, nb_batches=nb_batches
-        )
-    return merged_index
+        if index_optimizer is not None:
+            metric_infos = index_optimizer(merged_index, index_suffix=_generate_suffix(batch_id, nb_batches))
+        else:
+            metric_infos = None
+            _save_small_index(
+                index=merged_index, batch_id=batch_id, small_indices_folder=tmp_output_folder, nb_batches=nb_batches
+            )
+    else:
+        metric_infos = None
+    return merged_index, metric_infos
 
 
 def _get_file_system(path: str) -> fsspec.AbstractFileSystem:
     return fsspec.core.url_to_fs(path, use_listings_cache=False)[0]
 
 
-def _merge_to_n_indices(spark_session, n: int, src_folder: str, dst_folder: str):
+def _merge_to_n_indices(spark_session, n: int, src_folder: str, dst_folder: str, index_optimizer: Callable = None):
     """Merge all the indices from src_folder into n indices, and return the folder for the next stage"""
```

Comment:
We should update the comment for the return thing.

Reply:
done